templates/al2/runtime/bootstrap.sh

```diff
@@ -548,6 +548,10 @@ fi
 
 KUBELET_ARGS="$KUBELET_ARGS --cloud-provider=$KUBELET_CLOUD_PROVIDER"
 
+if vercmp "$KUBELET_VERSION" gteq "1.30.0"; then
+  KUBELET_ARGS="$KUBELET_ARGS --fail-swap-on=false --feature-gates=NodeSwap=false"
```

Comment:
We should do this in the config file instead of with flags, IMO. We also shouldn't change the default for `failSwapOn`; if the feature gate is disabled that param shouldn't matter?

Reply:
> default failSwapOn is true. so this flag does matter in terms of node join

That would only happen if the feature gate is enabled, right?

Reply:
I think we need it per this [doc](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/)
> Enable the kubelet to allocate swap memory for Kubernetes workloads on a node. Must be used with KubeletConfiguration.failSwapOn set to false. For more details, please see [swap memory](https://kubernetes.io/docs/concepts/architecture/nodes/#swap-memory)

Reply:
https://github.com/kubernetes/kubernetes/commit/32c265f60c175cedbd685f30e389c95fb1e66ee3

looks like this option has been around since at least 1.12, and it's never been tied to the `NodeSwap` feature gate 🤨.

Reply:
I guess we can set it to `false`, it just seems odd to set this option in such a way to support swap -- when we are explicitly *not* allowing swap

Reply:
yes, I feel this 2 level of feature flag control is odd.. 
above test I did was add 
```
  "featureGates": {
    "RotateKubeletServerCertificate": true,
    "NodeSwap": false
  },
```
but still run into the same issue.
running on t3 for another test

Reply:
> looks like this option has been around since at least 1.12, and it's never been tied to the `NodeSwap` feature gate 🤨.

experimental in 1.8 🙃 https://github.com/kubernetes/kubernetes/pull/47181/commits/71e8c8eba43a0fade6e4edfc739b331ba3cc658a

Reply:
lol touché!

I think it's safe to say that `c1.medium` and `m1.small` have never worked on EKS with AL2. I'm leaning "lets just document this and move on" vs touching the swap settings, which could have downstream impact on custom AMIs or other users of boostrap.sh (like canonical for Ubuntu)